src/XML/src/FMI3/fmi3_xml_variable.c

```diff
@@ -1083,14 +1063,21 @@ void fmi3_xml_variable_free_internals(jm_callbacks* callbacks, fmi3_xml_variable
             free_string_start_values(callbacks, var);
         }
     }
-    if (var->clocks) {
-        jm_vector_free(fmi3_value_reference_t)(var->clocks);
-        var->clocks = NULL;
+
+    if (var->aliases) {
+        size_t nAlias = jm_vector_get_size(jm_voidp)(var->aliases);
+        for (size_t i = 0; i < nAlias; i++) {
+            fmi3_xml_free_alias(callbacks, (fmi3_xml_alias_variable_t*)jm_vector_get_item(jm_voidp)(var->aliases, i));
+        }
+        jm_vector_free(jm_voidp)(var->aliases);
+        var->aliases = NULL;
     }
+
     if (var->clocks) {
-        callbacks->free(var->clocks);
```

Comment:
Diff looks weird, but essentially we tried to free clocks 2 times (but set it to NULL after first, so never happened), so I removed one.